net/udprelay/server.go

```diff
@@ -81,6 +81,63 @@ func newPairOfDiscoPubKeys(discoA, discoB key.DiscoPublic) pairOfDiscoPubKeys {
 	return pair
 }
 
+// ServerEndpointLamportID implements encoding.TextMarshaler and
+// encoding.TextUnmarshaler interfaces for consistent cross-platform
+// serialization behaviors.
+type ServerEndpointLamportID uint64
+
+func (s *ServerEndpointLamportID) UnmarshalText(text []byte) error {
+	parsed, err := strconv.ParseUint(string(text), 10, 64)
+	if err != nil {
+		return err
+	}
+	*s = ServerEndpointLamportID(parsed)
+	return nil
+}
+
+func (s ServerEndpointLamportID) MarshalText() ([]byte, error) {
+	return []byte(strconv.FormatUint(uint64(s), 10)), nil
+}
+
+// ServerEndpointVNI implements encoding.TextMarshaler and
+// encoding.TextUnmarshaler interfaces for consistent cross-platform
+// serialization behaviors.
+type ServerEndpointVNI uint32
+
+func (s *ServerEndpointVNI) UnmarshalText(text []byte) error {
+	parsed, err := strconv.ParseUint(string(text), 10, 24)
+	if err != nil {
+		return err
+	}
+	*s = ServerEndpointVNI(parsed)
+	return nil
+}
+
+func (s ServerEndpointVNI) MarshalText() ([]byte, error) {
+	if s > 1<<24-1 {
+		return nil, errors.New("vni exceeds 24 bits")
+	}
+	return []byte(strconv.FormatUint(uint64(s), 10)), nil
+}
+
+// ServerEndpointLifetime implements encoding.TextMarshaler and
+// encoding.TextUnmarshaler interfaces for consistent cross-platform
+// serialization behaviors.
+type ServerEndpointLifetime time.Duration
```

Comment:
If explicitly want to type that always uses the bespoke Go duration format, perhaps we should declare a `GoDuration` type or somethin in the "tstime" package?